command/services/export/export.go

```diff
@@ -0,0 +1,197 @@
+package export
+
+import (
+	"flag"
+	"fmt"
+	"strings"
+
+	"github.com/mitchellh/cli"
+
+	"github.com/hashicorp/consul/agent"
+	"github.com/hashicorp/consul/api"
+	"github.com/hashicorp/consul/command/flags"
+)
+
+func New(ui cli.Ui) *cmd {
+	c := &cmd{UI: ui}
+	c.init()
+	return c
+}
+
+type cmd struct {
+	UI    cli.Ui
+	flags *flag.FlagSet
+	http  *flags.HTTPFlags
+	help  string
+
+	serviceName    string
+	peerNames      string
+	partitionNames string
+}
+
+func (c *cmd) Run(args []string) int {
+	if err := c.flags.Parse(args); err != nil {
+		return 1
+	}
+
+	if err := c.validateFlags(); err != nil {
+		c.UI.Error(err.Error())
+		return 1
+	}
+
+	var peerNames []string
+	if c.peerNames != "" {
+		peerNames = strings.Split(c.peerNames, ",")
+		for _, peerName := range peerNames {
+			if peerName == "" {
+				c.UI.Error(fmt.Sprintf("Invalid peer %q", peerName))
+				return 1
+			}
+		}
+	}
+
+	partitionNames, err := c.getPartitionNames()
+	if err != nil {
+		c.UI.Error(err.Error())
+		return 1
+	}
+
+	client, err := c.http.APIClient()
+	if err != nil {
+		c.UI.Error(fmt.Sprintf("Error connect to Consul agent: %s", err))
+		return 1
+	}
+
+	entry, _, err := client.ConfigEntries().Get("exported-services", "default", nil)
+	if err != nil && !strings.Contains(err.Error(), agent.ConfigEntryNotFoundErr) {
+		c.UI.Error(fmt.Sprintf("Error reading config entry %s/%s: %v", "exported-services", "default", err))
+		return 1
+	}
+	if entry == nil {
+		cfg := api.ExportedServicesConfigEntry{
+			Name: "default",
+			Services: []api.ExportedService{
+				{
+					Name:      c.serviceName,
+					Consumers: buildConsumers(peerNames, partitionNames),
+				},
+			},
+		}
+		c.UI.Info(fmt.Sprintf("%#v", cfg))
+		_, _, err = client.ConfigEntries().Set(&cfg, &api.WriteOptions{})
+		if err != nil {
+			c.UI.Error(fmt.Sprintf("Error creating config entry: %s", err))
+			return 1
+		}
```

Comment:
Optional: instead of the many levels of nesting here, I might move the logic in the if statement out into helper functions like `CreateExportConfigEntry` and `UpdateExportConfigEntry`.